tests/test_matrixMap.py

```diff
@@ -90,6 +90,34 @@ def test_MatrixMapWithZeros(self):
         outdata2 = simplifiedMM.applyForward(indata)
         assert_allclose(outdata2, pred_outdata)
 
+    def test_MatrixMapShiftMapSimplify(self):
+        """Test that a non-square matrix map followed by a shift map can be simplified
+
+        This is ticket DM-10946
+        """
+        m1 = 1.0
+        m2 = 2.0
+        shift = 3.0
+        matrixMap = astshim.MatrixMap(np.array([[m1, m2]]))
+        self.assertEqual(matrixMap.nIn, 2)
+        self.assertEqual(matrixMap.nOut, 1)
+        shiftMap = astshim.ShiftMap([shift])
+        seriesMap = matrixMap.then(shiftMap)
+
+        indata = np.array([
+            [1.0, 2.0, 3.0],
+            [0.0, 1.0, 2.0],
+        ], dtype=float)
+        pred_outdata = np.array(m1 * indata[0] + m2 * indata[1] + shift)
+        pred_outdata.shape = (1, len(indata[0]))
```

Comment:
I did it this way to make the test as rigorous as possible. I agree it is more than the ticket requires, but I prefer to prove that a map does what it should do when possible. I tried to simplify the code slightly.

Also I realized the code should be in test_cmpMap.py so I moved it there. Then I discovered that was missing tests for `Mapping.then` and `Mapping.under` so I added those (as very simple additions to the existing test cases, with minor associated cleanup).